pilot/pkg/xds/ads.go

```diff
@@ -273,6 +269,11 @@ func (s *DiscoveryServer) Stream(stream DiscoveryStream) error {
 		log.Debugf("Unauthenticated XDS: %s", peerAddr)
 	}
 
+	if err := s.WaitForRequestLimit(stream.Context()); err != nil {
```

Comment:
Move this after authenticate

Reply:
? This assumes Authenticate is not a burden

Reply:
Yes. Also it does not make sense to wait for actual request limit, if the client is not authenticated.

Reply:
I have a concern if the auth calls kube-apiserver for token review

Reply:
I think request ratelimit was introduced for xds requests.

Reply:
ok. reverted that change